src/org/anjocaido/groupmanager/dataholder/WorldDataHolder.java

```diff
@@ -454,13 +453,11 @@ public boolean haveUsersChanged() {
 		if (users.isUsersChanged()) {
 			return true;
 		}
-		synchronized (users) {
-			for (User u : users.getUsers().values()) {
-				if (u.isChanged()) {
-					return true;

```

Comment:
I don't have an environment where I can do prolonged and extensive testing, so either this would need to be reverted or it would be published as a potentially highly unstable build